aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -65,7 +65,11 @@ namespace detail {
 struct DispatchKeyExtractor final {
 public:
   static DispatchKeyExtractor make(const FunctionSchema& schema) {
-    return DispatchKeyExtractor(schema.arguments().size());
+    return DispatchKeyExtractor(
+      schema.arguments().size(),
+      // TODO: replacing intType with scalarType and LayoutType after
+      // both type are defined correctly in jit_type.
+      getOptionalIndexForArgType_(IntType::get(), schema.arguments()));
```

Comment:
Note that in the final version (i.e. once it uses ScalarType instead of int), just looking for ScalarType is not enough. You also need to check that the other arguments (Layout, Device) are present and in the right position.